docs/concepts/simulation.md

```diff
@@ -0,0 +1,72 @@
+# Cosmos Blockchain Simulator
+
+The Cosmos SDK offers a full fledged simulation framework to fuzz test every message defined by a module.
+
+This functionality is provided by the[`SimApp`](https://github.com/cosmos/cosmos-sdk/blob/master/simapp/app.go),
+which is an dummy application that is used for running the [`simulation`](https://github.com/cosmos/cosmos-sdk/tree/master/x/simulation) module.
+This module defines all the simulation logic as well as the operations for randomized parameters like accounts, balances etc.
+
+## Goals
+
+The blockchain simulator tests how the blockchain application would behave under real life circumstances by generating and sending randomized messages.
+The goal of this is to detect and debug failures that could halt a live chain, by providing logs and statistics about the operations run by the simulator as well as exporting the latest application state when a failure was found.
+
+Its main difference with integration testing is that the simulator app allows you to pass parameters to customize the chain that's being simulated.
+This comes in handy when trying to reproduce bugs that were generated in the provided operations (randomized or not).
+
+## Simulation commands
+
+The simulation app has different commands, each of it tests a different failure type:
+
+- `FullAppSimulation`: General simulation mode. Runs the chain and the specified operations for a given number of blocks. Tests that there're no `panics` on the simulation.
+- `AppStateDeterminism`: Checks that all the nodes return the same values, in the same order.
+- `AppImportExport`: The simulator exportes the initial app state and then it creates a new app with the exported `genesis.json` as an input, checking for inconsistencies between the stores.
+- `AppSimulationAfterImport`: Queues two simulations together. The first one provides the app state (_i.e_ genesis) to the second. Useful to test software upgrades or hard-forks from a live chain.
+- `BenchmarkInvariants`: An invariant checks for differences between the values that are on the store and the passive tracker. Eg: total coins held by accounts vs total supply tracker.
```

Comment:
FullAppSimulation also does invariant checking right? Can you add how `BenchmarkInvariants` differs in that respect?